### ISSUE TITLE
Clients that use `getClientForDefaultRegion` and have unsupported regions in GovCloud result in hanging queries

### ISSUE THREAD

**Issue by cbruno10** [turbot/steampipe-plugin-aws#1593]
**Describe the bug**
For tables with service clients that are not available in all GovCloud regions and get the region from `getClientForDefaultRegion`, like CloudFront, Route 53, WAF, the queries will hang

**Steampipe version (`steampipe -v`)**
v0.18.3

**Plugin version (`steampipe plugin list`)**
v0.92.2

**To reproduce**
Setup an AWS connection for GovCloud and run `select * from aws_cloudfront_distribution;`

**Expected behavior**
Empty rows are returned

**Additional context**

**Comment by cbruno10** [comment]
@ParthaI In what other cases does AWS throw `no such host` errors?

Also, are we able to get a list of supported GovCloud regions for all AWS services that we can use to avoid making the API calls in the first place (similar to how we handle unsupported regions for commercial regions)? 

I think we should ultimately return no rows for these resources in unsupported GovCloud regions, like we do for commercial regions, but I'm worried about ignoring all `no such host` errors. Ignoring errors should only be used when we can't detect unsupported regions beforehand.

**Comment by ParthaI** [comment]
@cbruno10 Here is my `.spc` file for a gov cloud connection.

```
connection "aws_gov" {
  plugin  = "aws"
  regions = ["us-gov-*"]
  profile = "aws-gov-deadshot"
}
```

I have explored the usage of the global tables (`aws_cloudfront_distribution`, `aws_route53_domain`, `aws_route53_record`, and `aws_route53_zone`) which have the same endpoint, specifically `us-gov-west-1`.

Kindly note that AWS CloudFront Distribution is not supported in the Gov Cloud. The mentioned tables are intended for global AWS resources, and it is generally not within our scope to handle unsupported regions for global services.

Regarding the AWS Route53 tables, some of the resources are not supported in the Gov Cloud. For instance, `aws_route53_record` and `aws_route53_zone` do not throw any errors, but `aws_route53_domain` may encounter a `no such host` error.

Considering the scenario, I don't believe we can implement specific handling for unsupported regions in the global tables, as we have already addressed unsupported regions for commercial regions. I would appreciate any suggestions or alternative approaches to resolve this issue, as I have not identified any other solution beyond handling it during the API call itself.
